libs/domain/picking/src/components/picking/picking.component.ts

```diff
@@ -56,29 +145,117 @@ export class PickingComponent extends PickingListMixin(LitElement) {
+            tabs,
+            (tab) => html`<div
+              slot="panels"
+              id="tab-${tab.id}"
+              class="tab-panels"
+            >
+              ${when(
+                tab.items?.length,
+                () => html`
+                  ${repeat(
+                    tab.items,
+                    (item) => item.product.id,
+                    (item) =>
+                      html`
+                        <oryx-picking-product-card
+                          .productItem=${item}
+                          .status=${tab.id}
+                          @oryx.submit=${this.savePickingItem}
+                          @oryx.edit=${this.editPickingItem}
+                        ></oryx-picking-product-card>
+                      `
+                  )}
+                `,
+                () => this.renderFallback()
+              )}
+            </div>`
+          )}`
+      )}
+    `;
   }
 
   protected renderFallback(): TemplateResult {
-    return html`<p>${i18n('picking.no-picking-items-found')}</p>`;
+    const pickedItems = this.pickingList?.items.filter(
+      (item) => item.status === ItemsFilters.Picked
+    );
+
+    const isPickingComplete =
+      pickedItems?.length === this.pickingList?.items.length;
+
+    return html`
+      ${when(
+        isPickingComplete,
+        () =>
+          html`
+            <div class="picking-complete">
+              <div class="img-wrap">
+                <oryx-image resource="picking-items-processed"></oryx-image>
+              </div>
+              <h3 class="title-empty">${i18n(`picking.great-job`)}!</h3>
+              <p>${i18n(`picking.all-items-are-processed`)}!</p>
+            </div>
+          `
+      )}
+    `;
+  }
+
+  protected renderConfirmationModal(): TemplateResult {
+    return html`
+      <oryx-modal
+        ?open=${this.isConfirmPickingDialogOpen}
+        enableFooter
+        preventclosebybackdrop
+      >
+        <div slot="heading">
+          ${i18n('picking.product-card.confirm-picking')}
+        </div>
+
+        <span>
+          You only picked
+          <span class="bold-text">
+            ${this.partialPicking?.currentNumberOfPicked} out of
+            ${this.partialPicking?.quantity}
+          </span>
+          items. Do you really want to complete the pick?
+        </span>
+
+        <div slot="footer">
+          <oryx-button outline type="secondary">
+            <button @click=${this.onModalClose}>
+              ${i18n('picking.product-card.cancel')}
+            </button>
+          </oryx-button>
+
+          <oryx-button type="primary">
+            <button @click=${this.confirmPartialPicking}>
+              ${i18n('picking.product-card.confirm')}
+            </button>
+          </oryx-button>
+        </div>
+      </oryx-modal>
+    `;
+  }
+
+  private updatePickingItem(
+    productIndex: number,
+    numberOfPicked?: number
+  ): void {
+    this.items[productIndex].numberOfPicked = numberOfPicked ?? 0;
```

Comment:
Do items contain some unique identifier? Using index is not very safe

Reply:
Since the items are not being push or pop, i don't think there are high risk on this. Also we are providing type annotations for the callback function, that ensures typeScript infers the correct type and avoid potential type or runtime errors